types/react-navigation/index.d.ts

```diff
@@ -623,7 +624,7 @@ export interface NavigationEventSubscription {
 }
 
 export interface NavigationEventsProps extends ViewProps {
-  navigation?: NavigationNavigator;
```

Comment:
I believe this was incorrect before... `NavigationEvents` should have a `navigation.addListener` prop which only exists on `NavigationScreenProp`.